pkg/cloud/cloud_io.go

```diff
@@ -231,6 +231,7 @@ func (r *ResumingReader) Read(p []byte) (int, error) {
 			}
 			log.Errorf(r.Ctx, "Retry IO: error %s", lastErr)
 			lastErr = nil
+			r.Reader.Close()
```

Comment:
hmm actually i think we need a nil check here incase the call to `r.Open()` failed.